### ISSUE TITLE
provide ts_tree_get_changed_ranges binding

### ISSUE THREAD

**Issue by coiby** [tree-sitter/py-tree-sitter#28]
It would be great to expose this `ts_tree_get_changed_ranges binding` for Python

```c
/**
 * Compare an old edited syntax tree to a new syntax tree representing the same
 * document, returning an array of ranges whose syntactic structure has changed.
 *
 * For this to work correctly, the old syntax tree must have been edited such
 * that its ranges match up to the new tree. Generally, you'll want to call
 * this function right after calling one of the `ts_parser_parse` functions.
 * You need to pass the old tree that was passed to parse, as well as the new
 * tree that was returned from that function.
 *
 * The returned array is allocated using `malloc` and the caller is responsible
 * for freeing it using `free`. The length of the array will be written to the
 * given `length` pointer.
 */
TSRange *ts_tree_get_changed_ranges(
  const TSTree *old_tree,
  const TSTree *new_tree,
  uint32_t *length
);
```
https://github.com/tree-sitter/tree-sitter/blob/master/lib/include/tree_sitter/api.h

**Comment by meakbiyik** [comment]
I second this. @maxbrunsfeld is py-tree-sitter not maintained anymore? There hasn't been any additions for some time and that bug in #18 just messes with the bindings. This would be the most commonly used tree-sitter binding if that issue were to be corrected 🌞

**Comment by lunixbochs** [comment]
implemented, see #56 and 3ce2e23df2b109e2fb3ba047ec76c38b4073cc47

**Comment by coiby** [comment]
Thanks for the implementation!
